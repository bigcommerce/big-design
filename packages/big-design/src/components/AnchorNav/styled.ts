import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
import styled from 'styled-components';

import { BoxProps } from '../Box';

export const StyledAnchorNav = styled.div<BoxProps>`
display: block;
position relative;

.anchor-nav {
    &__list {
        position: sticky;
        inset-block-start: 0;
        width: 266px;
        float: inline-start;
        display:none;

        @media (min-width: ${({ theme }) => theme.breakpointValues.tablet}) {
            display: block;
            margin-inline-end: ${defaultTheme.spacing.xLarge};
        }

        & > ul {
            list-style-type: none;
            padding: 0;
            margin: 0;

            & > li {
                display:block;
                margin: 0;
                padding: 0;
                box-sizing: border-box;

                & > a {
                    color: ${defaultTheme.colors.primary};
                    text-decoration: none;
                    display: block;
                    padding: ${defaultTheme.spacing.small} ${defaultTheme.spacing.large};
                    border-inline-start: 3px solid transparent;

                    &:hover {
                        background-color: ${defaultTheme.colors.primary10};
                    }

                    &.active {
                        color: ${defaultTheme.colors.secondary70};
                        border-inline-start: 3px solid ${defaultTheme.colors.primary};

                        &:hover {
                            background-color: transparent;
                        }
                    }
                }
            }
        }
    
    }
    
    &__elements {
        display:flex;
        flex-direction: column;
        gap: ${defaultTheme.spacing.xLarge};
    }
}
`;

StyledAnchorNav.defaultProps = { theme: defaultTheme };
