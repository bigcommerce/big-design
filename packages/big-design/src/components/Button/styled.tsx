import { addValues, theme as defaultTheme } from '@bigcommerce/big-design-theme';
import styled, { css } from 'styled-components';

import { MarginProps, withMargins } from '../../helpers';
import { withTransition } from '../../helpers/transitions';
import { Flex } from '../Flex';

import { ButtonProps } from './index';

export const StyledButton = styled.button<ButtonProps & MarginProps>`
  ${withTransition(['background-color', 'border-color', 'box-shadow', 'color'])}

  && {
    ${withMargins()};
  }

  align-items: center;
  appearance: none;
  border: ${({ theme }) => theme.border.box};
  border-radius: ${({ theme }) => theme.borderRadius.normal};
  color: ${({ theme }) => theme.colors.white};
  cursor: pointer;
  display: inline-flex;
  flex: none;
  font-size: ${({ theme }) => theme.typography.fontSize.medium};
  font-weight: ${({ theme }) => theme.typography.fontWeight.regular};
  height: ${({ theme }) => addValues(theme.spacing.xxLarge, theme.spacing.xxSmall)};
  justify-content: center;
  line-height: ${({ theme }) => theme.lineHeight.xLarge};
  outline: none;
  padding: ${({ theme }) => `0 ${theme.spacing.medium}`};
  position: relative;
  text-align: center;
  text-decoration: none;
  user-select: none;
  vertical-align: middle;
  white-space: nowrap;
  width: ${({ mobileWidth }) => (mobileWidth === 'auto' ? 'auto' : '100%')};

  &:focus {
    outline: none;
  }

  &[disabled] {
    border-color: ${({ theme }) => theme.colors.secondary30};
    pointer-events: none;
  }

  & + .bd-button {
    margin-top: ${({ mobileWidth, theme }) => mobileWidth === '100%' && theme.spacing.xSmall};
    margin-left: ${({ mobileWidth, theme }) => mobileWidth === 'auto' && theme.spacing.xSmall};

    ${({ theme }) => theme.breakpoints.tablet} {
      margin-top: ${({ theme }) => theme.spacing.none};
      margin-left: ${({ theme }) => theme.spacing.xSmall};
    }
  }

  ${({ theme }) => theme.breakpoints.tablet} {
    width: auto;

    ${({ iconOnly: icon, theme }) =>
      icon &&
      css`
        padding: 0;
        min-width: ${addValues(theme.spacing.xxLarge, theme.spacing.xxSmall)};
      `};
  }

  ${({ iconLeft, theme }) =>
    iconLeft &&
    css`
      padding-left: ${theme.spacing.xSmall};
    `};

  ${({ iconRight, theme }) =>
    iconRight &&
    css`
      padding-right: ${theme.spacing.xSmall};
    `};

  ${(props) => getButtonStyles(props)}
`;

export const ContentWrapper = styled.span.attrs<Record<string, unknown>, { isLoading?: boolean }>(
  {},
)`
  align-content: center;
  align-items: center;
  display: inline-grid;
  grid-auto-flow: column;
  grid-gap: ${({ theme }) => theme.spacing.xSmall};

  ${({ isLoading }) =>
    isLoading &&
    css`
      visibility: hidden;
    `};
`;

export const LoadingSpinnerWrapper = styled(Flex)`
  position: absolute;
`;

/**
 * These can be generated dynamically but I'm leaning towards being extra
 * explicit and being able to handle corner cases and changes from design easily
 */
const ButtonPrimary = css<ButtonProps>`
  background-color: ${({ theme }) => theme.colors.primary};
  border-color: ${({ theme }) => theme.colors.primary};
  font-weight: ${({ theme }) => theme.typography.fontWeight.semiBold};

  &:active {
    background-color: ${({ theme }) => theme.colors.primary60};
  }

  &:focus {
    box-shadow: ${({ theme }) => `0 0 0 ${theme.spacing.xxSmall} ${theme.colors.primary20}`};
  }

  &:hover:not(:active) {
    background-color: ${({ theme }) => theme.colors.primary50};
  }

  &[disabled] {
    background-color: ${({ theme }) => theme.colors.secondary30};
  }
`;

const ButtonPrimaryDestructive = css<ButtonProps>`
  background-color: ${({ theme }) => theme.colors.danger};
  border-color: ${({ theme }) => theme.colors.danger};
  font-weight: ${({ theme }) => theme.typography.fontWeight.semiBold};

  &:active {
    background-color: ${({ theme }) => theme.colors.danger60};
  }

  &:focus {
    box-shadow: ${({ theme }) => `0 0 0 ${theme.spacing.xxSmall} ${theme.colors.danger20}`};
  }

  &:hover:not(:active) {
    background-color: ${({ theme }) => theme.colors.danger50};
  }

  &[disabled] {
    background-color: ${({ theme }) => theme.colors.secondary30};
  }
`;

const ButtonSecondary = css<ButtonProps>`
  background-color: transparent;
  border-color: ${({ theme }) => theme.colors.primary};
  color: ${({ theme }) => theme.colors.primary};

  &:active {
    background-color: ${({ theme }) => theme.colors.primary20};
  }

  &:focus {
    box-shadow: ${({ theme }) => `0 0 0 ${theme.spacing.xxSmall} ${theme.colors.primary20}`};
  }

  &:hover:not(:active) {
    background-color: ${({ theme }) => theme.colors.primary10};
  }

  &[disabled] {
    color: ${({ theme }) => theme.colors.secondary30};
  }
`;

const ButtonSecondaryDestructive = css<ButtonProps>`
  background-color: transparent;
  border-color: ${({ theme }) => theme.colors.danger};
  color: ${({ theme }) => theme.colors.danger};

  &:active {
    background-color: ${({ theme }) => theme.colors.danger20};
  }

  &:focus {
    box-shadow: ${({ theme }) => `0 0 0 ${theme.spacing.xxSmall} ${theme.colors.danger20}`};
  }

  &:hover:not(:active) {
    background-color: ${({ theme }) => theme.colors.danger10};
  }

  &[disabled] {
    color: ${({ theme }) => theme.colors.secondary30};
  }
`;

const ButtonSubtle = css<ButtonProps>`
  background-color: transparent;
  border-color: transparent;
  color: ${({ theme }) => theme.colors.primary};

  &:active {
    background-color: ${({ theme }) => theme.colors.primary20};
  }

  &:focus {
    box-shadow: ${({ theme }) => `0 0 0 ${theme.spacing.xxSmall} ${theme.colors.primary20}`};
  }

  &:hover:not(:active) {
    background-color: ${({ theme }) => theme.colors.primary10};
  }

  &[disabled] {
    border-color: transparent;
    color: ${({ theme }) => theme.colors.secondary30};
  }
`;

const ButtonSubtleDestructive = css<ButtonProps>`
  background-color: transparent;
  border-color: transparent;
  color: ${({ theme }) => theme.colors.danger};

  &:active {
    background-color: ${({ theme }) => theme.colors.danger20};
  }

  &:focus {
    box-shadow: ${({ theme }) => `0 0 0 ${theme.spacing.xxSmall} ${theme.colors.danger20}`};
  }

  &:hover:not(:active) {
    background-color: ${({ theme }) => theme.colors.danger10};
  }

  &[disabled] {
    border-color: transparent;
    color: ${({ theme }) => theme.colors.secondary30};
  }
`;

const ButtonUtility = css<ButtonProps>`
  background-color: transparent;
  border-color: transparent;
  color: ${({ theme }) => theme.colors.secondary60};
  padding-inline: ${({ theme }) => theme.spacing.xxSmall};
  min-width: ${({ theme }) => addValues(theme.spacing.xLarge, theme.spacing.small)};

  &:active {
    background-color: ${({ theme }) => theme.colors.primary20};
    color: ${({ theme }) => theme.colors.primary60};
  }

  &:focus:not(:active) {
    box-shadow: ${({ theme }) => `0 0 0 ${theme.spacing.xxSmall} ${theme.colors.secondary10}`};
    color: ${({ theme }) => theme.colors.primary50};
  }

  &:hover:not(:active) {
    background-color: ${({ theme }) => theme.colors.primary10};
    color: ${({ theme }) => theme.colors.primary50};
  }

  &[disabled] {
    border-color: transparent;
    color: ${({ theme }) => theme.colors.secondary50};
  }
`;

const ButtonUtilityDestructive = css<ButtonProps>`
  background-color: transparent;
  border-color: transparent;
  color: ${({ theme }) => theme.colors.secondary60};

  &:active {
    background-color: ${({ theme }) => theme.colors.danger20};
    color: ${({ theme }) => theme.colors.danger60};
  }

  &:focus:not(:active) {
    box-shadow: ${({ theme }) => `0 0 0 ${theme.spacing.xxSmall} ${theme.colors.danger20}`};
    color: ${({ theme }) => theme.colors.danger40};
  }

  &:hover:not(:active) {
    background-color: ${({ theme }) => theme.colors.danger10};
    color: ${({ theme }) => theme.colors.danger50};
  }

  &[disabled] {
    border-color: transparent;
    color: ${({ theme }) => theme.colors.secondary50};
  }
`;

function getButtonStyles(props: ButtonProps) {
  const { actionType, variant } = props;

  switch (variant) {
    case 'primary':
      return actionType === 'destructive' ? ButtonPrimaryDestructive : ButtonPrimary;

    case 'secondary':
      return actionType === 'destructive' ? ButtonSecondaryDestructive : ButtonSecondary;

    case 'subtle':
      return actionType === 'destructive' ? ButtonSubtleDestructive : ButtonSubtle;

    case 'utility':
      return actionType === 'destructive' ? ButtonUtilityDestructive : ButtonUtility;
  }
}

StyledButton.defaultProps = { theme: defaultTheme };
ContentWrapper.defaultProps = { theme: defaultTheme };
LoadingSpinnerWrapper.defaultProps = { theme: defaultTheme };
