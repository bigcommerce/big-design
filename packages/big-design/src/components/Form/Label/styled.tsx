import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
import styled, { DefaultTheme, StyledComponent } from 'styled-components';

import { StyleableH4 } from '../../Typography/private';
import { HeadingProps } from '../../Typography/types';

import { type FormControlLabelProps } from './Label';

interface StyledLabelArgument extends Omit<FormControlLabelProps, 'localization'> {
  theme: DefaultTheme;
}

export const StyledLabel = styled<
  StyledComponent<'label' | 'h4', DefaultTheme, Partial<HeadingProps>> & FormControlLabelProps
>(StyleableH4).attrs({
  as: 'label',
})<FormControlLabelProps>`
  cursor: pointer;
  display: inline-block;
  margin-bottom: ${({ theme }: StyledLabelArgument) => theme.spacing.xxSmall};
`;

StyledLabel.defaultProps = { theme: defaultTheme };
