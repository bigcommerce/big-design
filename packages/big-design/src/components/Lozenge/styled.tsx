import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
import styled, { css } from 'styled-components';

import { Box } from '../Box';

import { LozengeProps } from './Lozenge';

interface StyledLozengeProps extends Omit<LozengeProps, 'label' | 'tooltipContent'> {
  hasTooltip?: boolean;
}

const variantStyles = {
  alpha: css`
    background-color: ${({ theme }) => theme.colors.warning20};
    color: ${({ theme }) => theme.colors.secondary70};
  `,
  beta: css`
    background-color: ${({ theme }) => theme.colors.primary20};
    color: ${({ theme }) => theme.colors.primary50};
  `,
  deprecated: css`
    background-color: ${({ theme }) => theme.colors.danger20};
    color: ${({ theme }) => theme.colors.danger70};
  `,
  legacy: css`
    background-color: ${({ theme }) => theme.colors.secondary30};
    color: ${({ theme }) => theme.colors.secondary70};
  `,
  new: css`
    background-color: ${({ theme }) => theme.colors.success20};
    color: ${({ theme }) => theme.colors.success70};
  `,
};

export const StyledLozenge = styled(Box)<StyledLozengeProps>`
  border-radius: ${({ theme }) => theme.spacing.medium};
  display: inline-flex;
  font-size: ${({ theme }) => theme.typography.fontSize.small};
  gap: ${({ theme }) => theme.spacing.xxSmall};
  line-height: ${({ theme }) => theme.lineHeight.small};
  text-align: center;
  padding-block: ${({ theme }) => theme.spacing.xxSmall};
  vertical-align: middle;
  padding-inline: ${({ theme, hasTooltip }) =>
    hasTooltip ? `${theme.spacing.small} ${theme.spacing.xxSmall}` : theme.spacing.small};
  white-space: nowrap;

  ${({ variant = 'new' }) => variantStyles[variant]};
`;

StyledLozenge.defaultProps = {
  theme: defaultTheme,
  hasTooltip: false,
};
