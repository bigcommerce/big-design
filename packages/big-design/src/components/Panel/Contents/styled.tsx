import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
import styled, { css } from 'styled-components';

import { Box } from '../../Box';

import { PanelContentProps } from './Contents';

export const StyledPanelContentsWrapper = styled(Box)<Omit<PanelContentProps, 'children'>>`
  ${({ theme, scrollable }) =>
    scrollable &&
    css`
      background-image: linear-gradient(
          0deg,
          ${theme.colors.secondary40} 0%,
          transparent 1px,
          transparent 100%
        ),
        linear-gradient(180deg, ${theme.colors.secondary40} 0%, transparent 1px, transparent 100%);
      overflow: auto;
    `}

  ${({ height }) =>
    height &&
    css`
      height: ${height};
    `}

  ${({ theme, padded }) =>
    padded !== true &&
    css`
      margin-inline: -${theme.spacing.medium};
    `}

  ${({ theme }) => theme.breakpoints.tablet} {
    ${({ theme, padded }) =>
      padded !== true &&
      css`
        margin-inline: -${theme.spacing.xLarge};
      `}
  }
`;
StyledPanelContentsWrapper.defaultProps = {
  theme: defaultTheme,
  padded: true,
  scrollable: false,
  height: 'auto',
};

export const StyledPanelContents = styled(Box)<Omit<PanelContentProps, 'children'>>`
  min-height: 100%;
  ${({ theme, scrollable }) =>
    scrollable &&
    css`
      border-block-start: ${theme.border.box};
      border-block-end: ${theme.border.box};

      border-block-start-color: ${theme.colors.white};
      border-block-end-color: ${theme.colors.white};
    `}
`;
StyledPanelContents.defaultProps = {
  theme: defaultTheme,
  padded: true,
  scrollable: false,
  height: 'auto',
};
