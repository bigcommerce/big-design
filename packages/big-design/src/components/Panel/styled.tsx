import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
import styled from 'styled-components';

import { Box } from '../Box';
import { StyleableH2 } from '../Typography/private';

export const StyledPanel = styled(Box)`
  border-radius: ${({ theme }) => theme.borderRadius.none};

  ${({ theme }) => theme.breakpoints.tablet} {
    border-radius: ${({ theme }) => theme.borderRadius.normal};
  }
`;

export const StyledH2 = styled(StyleableH2)`
  flex-grow: 1;
  flex-shrink: 1;

  & ~ .bd-button {
    width: auto;
    margin-top: 0;
  }
`;

StyledPanel.defaultProps = { theme: defaultTheme };
StyledH2.defaultProps = { theme: defaultTheme };
