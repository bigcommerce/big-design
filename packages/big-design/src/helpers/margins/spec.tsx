import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
import 'jest-styled-components';
import { render, screen } from '@testing-library/react';
import React from 'react';
import styled from 'styled-components';

import { excludeMarginProps, MarginProps, withMargins } from './margins';

const TestComponent = styled.div<MarginProps>`
  ${withMargins()};
`;

TestComponent.defaultProps = { theme: defaultTheme };

test('margin', () => {
  render(<TestComponent data-testid="test-component" margin="medium" />);

  expect(screen.getByTestId('test-component')).toHaveStyle('margin: 1rem');
});

test('marginTop', () => {
  render(<TestComponent data-testid="test-component" marginTop="medium" />);

  expect(screen.getByTestId('test-component')).toHaveStyle('margin-top: 1rem');

  expect(screen.getByTestId('test-component')).not.toHaveStyle('margin: 1rem');
  expect(screen.getByTestId('test-component')).not.toHaveStyle('margin-right: 1rem');
  expect(screen.getByTestId('test-component')).not.toHaveStyle('margin-left: 1rem');
  expect(screen.getByTestId('test-component')).not.toHaveStyle('margin-bottom: 1rem');
});

test('marginRight', () => {
  render(<TestComponent data-testid="test-component" marginRight="medium" />);

  expect(screen.getByTestId('test-component')).toHaveStyle('margin-right: 1rem');

  expect(screen.getByTestId('test-component')).not.toHaveStyle('margin: 1rem');
  expect(screen.getByTestId('test-component')).not.toHaveStyle('margin-top: 1rem');
  expect(screen.getByTestId('test-component')).not.toHaveStyle('margin-left: 1rem');
  expect(screen.getByTestId('test-component')).not.toHaveStyle('margin-bottom: 1rem');
});

test('marginBottom', () => {
  render(<TestComponent data-testid="test-component" marginBottom="medium" />);

  expect(screen.getByTestId('test-component')).toHaveStyle('margin-bottom: 1rem');

  expect(screen.getByTestId('test-component')).not.toHaveStyle('margin: 1rem');
  expect(screen.getByTestId('test-component')).not.toHaveStyle('margin-top: 1rem');
  expect(screen.getByTestId('test-component')).not.toHaveStyle('margin-right: 1rem');
  expect(screen.getByTestId('test-component')).not.toHaveStyle('margin-left: 1rem');
});

test('marginLeft', () => {
  render(<TestComponent data-testid="test-component" marginLeft="medium" />);

  expect(screen.getByTestId('test-component')).toHaveStyle('margin-left: 1rem');

  expect(screen.getByTestId('test-component')).not.toHaveStyle('margin: 1rem');
  expect(screen.getByTestId('test-component')).not.toHaveStyle('margin-top: 1rem');
  expect(screen.getByTestId('test-component')).not.toHaveStyle('margin-right: 1rem');
  expect(screen.getByTestId('test-component')).not.toHaveStyle('margin-bottom: 1rem');
});

test('marginVertical', () => {
  render(<TestComponent data-testid="test-component" marginVertical="medium" />);

  expect(screen.getByTestId('test-component')).toHaveStyle('margin-top: 1rem');
  expect(screen.getByTestId('test-component')).toHaveStyle('margin-bottom: 1rem');

  expect(screen.getByTestId('test-component')).not.toHaveStyle('margin: 1rem');
  expect(screen.getByTestId('test-component')).not.toHaveStyle('margin-right: 1rem');
  expect(screen.getByTestId('test-component')).not.toHaveStyle('margin-left: 1rem');
});

test('marginHorizontal', () => {
  render(<TestComponent data-testid="test-component" marginHorizontal="medium" />);

  expect(screen.getByTestId('test-component')).toHaveStyle('margin-left: 1rem');
  expect(screen.getByTestId('test-component')).toHaveStyle('margin-right: 1rem');

  expect(screen.getByTestId('test-component')).not.toHaveStyle('margin: 1rem');
  expect(screen.getByTestId('test-component')).not.toHaveStyle('margin-top: 1rem');
  expect(screen.getByTestId('test-component')).not.toHaveStyle('margin-bottom: 1rem');
});

test('simple margins combination', () => {
  render(
    <TestComponent
      data-testid="test-component"
      marginBottom="none"
      marginRight="small"
      marginTop="medium"
    />,
  );

  expect(screen.getByTestId('test-component')).toHaveStyle('margin-top: 1rem');
  expect(screen.getByTestId('test-component')).toHaveStyle('margin-bottom: 0px');
  expect(screen.getByTestId('test-component')).toHaveStyle('margin-right: 0.75rem');
});

test('responsive margin', () => {
  render(
    <TestComponent
      data-testid="test-component"
      margin={{ mobile: 'none', tablet: 'small', desktop: 'medium' }}
    />,
  );

  expect(screen.getByTestId('test-component')).toMatchSnapshot();
});

test('responsive marginTop', () => {
  render(
    <TestComponent
      data-testid="test-component"
      marginTop={{ mobile: 'none', tablet: 'small', desktop: 'medium' }}
    />,
  );

  expect(screen.getByTestId('test-component')).toMatchSnapshot();
});

test('responsive marginRight', () => {
  render(
    <TestComponent
      data-testid="test-component"
      marginRight={{ mobile: 'none', tablet: 'small', desktop: 'medium' }}
    />,
  );

  expect(screen.getByTestId('test-component')).toMatchSnapshot();
});

test('responsive marginBottom', () => {
  render(
    <TestComponent
      data-testid="test-component"
      marginBottom={{ mobile: 'none', tablet: 'small', desktop: 'medium' }}
    />,
  );

  expect(screen.getByTestId('test-component')).toMatchSnapshot();
});

test('responsive marginLeft', () => {
  render(
    <TestComponent
      data-testid="test-component"
      marginLeft={{ mobile: 'none', tablet: 'small', desktop: 'medium' }}
    />,
  );

  expect(screen.getByTestId('test-component')).toMatchSnapshot();
});

test('responsive marginVertical', () => {
  render(
    <TestComponent
      data-testid="test-component"
      marginVertical={{ mobile: 'none', tablet: 'small', desktop: 'medium' }}
    />,
  );

  expect(screen.getByTestId('test-component')).toMatchSnapshot();
});

test('responsive marginHorizontal', () => {
  render(
    <TestComponent
      data-testid="test-component"
      marginHorizontal={{ mobile: 'none', tablet: 'small', desktop: 'medium' }}
    />,
  );

  expect(screen.getByTestId('test-component')).toMatchSnapshot();
});

test('responsive and non responsive combination', () => {
  render(
    <TestComponent
      data-testid="test-component"
      marginBottom="medium"
      marginLeft={{ mobile: 'none', tablet: 'small', desktop: 'medium' }}
      marginTop="none"
    />,
  );

  expect(screen.getByTestId('test-component')).toMatchSnapshot();
});

test('excludeMarginProps removes all margin props from an object', () => {
  const props: MarginProps & { testKey: 'test-value' } = {
    testKey: 'test-value',
    margin: 'medium',
    marginTop: 'medium',
    marginRight: 'medium',
    marginBottom: 'medium',
    marginLeft: 'medium',
    marginVertical: 'medium',
    marginHorizontal: 'medium',
  };

  expect(excludeMarginProps(props)).toEqual({ testKey: 'test-value' });
});

test('non-spacing value auto', () => {
  render(<TestComponent data-testid="test-component" margin="auto" />);

  expect(screen.getByTestId('test-component')).toHaveStyle('margin: auto');
});
