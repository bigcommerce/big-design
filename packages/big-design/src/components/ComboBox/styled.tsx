import { ArrowDropDownIcon } from '@bigcommerce/big-design-icons';
import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
import { hideVisually } from 'polished';
import styled, { css } from 'styled-components';

import { StyleableButton } from '../Button/private';

export const StyledStatusMessage = styled.div`
  ${hideVisually()}
`;

export const StyledDropdownIcon = styled(ArrowDropDownIcon)<{ disabled?: boolean }>`
  ${({ disabled }) =>
    !disabled &&
    css`
      :hover {
        cursor: pointer;
      }
    `}
`;

export const StyledInputContainer = styled.div`
  input[readonly] {
    cursor: pointer;
  }
`;

export const DropdownButton = styled(StyleableButton)`
  color: ${({ theme }) => theme.colors.secondary60};
  height: auto;
  padding: 0;
`;

StyledStatusMessage.defaultProps = { theme: defaultTheme };
StyledDropdownIcon.defaultProps = { theme: defaultTheme };
StyledInputContainer.defaultProps = { theme: defaultTheme };
DropdownButton.defaultProps = { theme: defaultTheme };
