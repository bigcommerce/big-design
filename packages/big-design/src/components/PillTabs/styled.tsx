import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
import styled, { css } from 'styled-components';

import { StyleableButton } from '../Button/private';
import { FlexItem } from '../Flex';

interface StyledPillTabProps {
  isActive: boolean;
}

interface StyledFlexItemProps {
  isVisible: boolean;
  separator: boolean;
}

export const StyledPillTab = styled(StyleableButton)<StyledPillTabProps>`
  ${(props) =>
    props.isActive &&
    css`
      background-color: ${({ theme }) => theme.colors.primary20};
    `}
`;

export const StyledFlexItem = styled(FlexItem)<StyledFlexItemProps>`
  ${(props) =>
    !props.isVisible &&
    css`
      position: absolute;
      visibility: hidden;
      z-index: ${({ theme }) => -theme.zIndex.tooltip};
    `}

  ${(props) =>
    props.separator &&
    props.isVisible &&
    css`
      border-left: 1px solid ${({ theme }) => theme.colors.secondary30};
      padding-left: ${({ theme }) => theme.spacing.xSmall};
    `}
`;

StyledPillTab.defaultProps = { theme: defaultTheme };
StyledFlexItem.defaultProps = { theme: defaultTheme };
