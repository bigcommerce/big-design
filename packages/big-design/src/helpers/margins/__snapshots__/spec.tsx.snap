// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`responsive and non responsive combination 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 1rem;
}

@media (min-width:0px) {
  .c0 {
    margin-left: 0;
  }
}

@media (min-width:720px) {
  .c0 {
    margin-left: 0.75rem;
  }
}

@media (min-width:1025px) {
  .c0 {
    margin-left: 1rem;
  }
}

<div
  class="c0"
/>
`;

exports[`responsive margin 1`] = `
@media (min-width:0px) {
  .c0 {
    margin: 0;
  }
}

@media (min-width:720px) {
  .c0 {
    margin: 0.75rem;
  }
}

@media (min-width:1025px) {
  .c0 {
    margin: 1rem;
  }
}

<div
  class="c0"
/>
`;

exports[`responsive marginBottom 1`] = `
@media (min-width:0px) {
  .c0 {
    margin-bottom: 0;
  }
}

@media (min-width:720px) {
  .c0 {
    margin-bottom: 0.75rem;
  }
}

@media (min-width:1025px) {
  .c0 {
    margin-bottom: 1rem;
  }
}

<div
  class="c0"
/>
`;

exports[`responsive marginHorizontal 1`] = `
@media (min-width:0px) {
  .c0 {
    margin-left: 0;
    margin-right: 0;
  }
}

@media (min-width:720px) {
  .c0 {
    margin-left: 0.75rem;
    margin-right: 0.75rem;
  }
}

@media (min-width:1025px) {
  .c0 {
    margin-left: 1rem;
    margin-right: 1rem;
  }
}

<div
  class="c0"
/>
`;

exports[`responsive marginLeft 1`] = `
@media (min-width:0px) {
  .c0 {
    margin-left: 0;
  }
}

@media (min-width:720px) {
  .c0 {
    margin-left: 0.75rem;
  }
}

@media (min-width:1025px) {
  .c0 {
    margin-left: 1rem;
  }
}

<div
  class="c0"
/>
`;

exports[`responsive marginRight 1`] = `
@media (min-width:0px) {
  .c0 {
    margin-right: 0;
  }
}

@media (min-width:720px) {
  .c0 {
    margin-right: 0.75rem;
  }
}

@media (min-width:1025px) {
  .c0 {
    margin-right: 1rem;
  }
}

<div
  class="c0"
/>
`;

exports[`responsive marginTop 1`] = `
@media (min-width:0px) {
  .c0 {
    margin-top: 0;
  }
}

@media (min-width:720px) {
  .c0 {
    margin-top: 0.75rem;
  }
}

@media (min-width:1025px) {
  .c0 {
    margin-top: 1rem;
  }
}

<div
  class="c0"
/>
`;

exports[`responsive marginVertical 1`] = `
@media (min-width:0px) {
  .c0 {
    margin-top: 0;
    margin-bottom: 0;
  }
}

@media (min-width:720px) {
  .c0 {
    margin-top: 0.75rem;
    margin-bottom: 0.75rem;
  }
}

@media (min-width:1025px) {
  .c0 {
    margin-top: 1rem;
    margin-bottom: 1rem;
  }
}

<div
  class="c0"
/>
`;
